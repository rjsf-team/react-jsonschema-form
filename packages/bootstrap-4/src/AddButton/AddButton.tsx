import React from "react";

import { AddButtonProps } from "@rjsf/core";
import Button from "react-bootstrap/Button";
import { BsPlus } from "react-icons/bs";

const AddButton: React.FC<AddButtonProps> = props => (
  <Button {...props} color="primary" style={{width: "100%"}} className="ml-1">
    <BsPlus/>
  </Button>
);

export default AddButton;
