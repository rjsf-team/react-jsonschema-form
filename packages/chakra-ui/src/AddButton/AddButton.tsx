import { FormContextType, IconButtonProps, RJSFSchema, StrictRJSFSchema, TranslatableString } from '@rjsf/utils';
import { Button } from '@chakra-ui/react';
import { AddIcon } from '@chakra-ui/icons';

export default function AddButton<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any>({
  uiSchema,
  registry,
  ...props
}: IconButtonProps<T, S, F>) {
  const { translateString } = registry;
  return (
    <Button leftIcon={<AddIcon />} {...props}>
      {translateString(TranslatableString.AddItemButton)}
    </Button>
  );
}
