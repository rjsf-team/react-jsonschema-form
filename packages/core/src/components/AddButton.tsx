import React from "react";

import IconButton from "./IconButton";

export interface AddButtonProps {
  className?: string;
  onClick?: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
  disabled?: boolean;
}

export default function AddButton({
  className,
  onClick,
  disabled,
}: AddButtonProps) {
  return (
    <div className="row">
      <p className={`col-xs-3 col-xs-offset-9 text-right ${className}`}>
        <IconButton
          type="info"
          icon="plus"
          className="btn-add col-xs-12"
          aria-label="Add"
          tabIndex="0"
          onClick={onClick}
          disabled={disabled}
        />
      </p>
    </div>
  );
}
