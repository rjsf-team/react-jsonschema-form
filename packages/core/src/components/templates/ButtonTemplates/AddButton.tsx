import { FormContextType, IconButtonProps, RJSFSchema, StrictRJSFSchema, TranslatableString } from '@rjsf/utils';

import IconButton from './IconButton';

/** The `AddButton` renders a button that represent the `Add` action on a form
 */
export default function AddButton<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any>({
  className,
  onClick,
  disabled,
  registry,
}: IconButtonProps<T, S, F>) {
  const { translateString } = registry;
  return (
    <div className='row'>
      <p className={`col-xs-3 col-xs-offset-9 text-right ${className}`}>
        <IconButton
          iconType='info'
          icon='plus'
          className='btn-add col-xs-12'
          title={translateString(TranslatableString.AddButton)}
          onClick={onClick}
          disabled={disabled}
          registry={registry}
        />
      </p>
    </div>
  );
}
