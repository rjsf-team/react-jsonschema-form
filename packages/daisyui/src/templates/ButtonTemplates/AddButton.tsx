import { FormContextType, IconButtonProps, RJSFSchema, StrictRJSFSchema, TranslatableString } from '@rjsf/utils';
import { library } from '@fortawesome/fontawesome-svg-core';
import { faArrowDown, faArrowUp, faCopy, faTrash, faPlus } from '@fortawesome/free-solid-svg-icons';
import type { IconDefinition } from '@fortawesome/fontawesome-svg-core';

import DaisyUIButton from './DaisyUIButton';

library.add(faPlus, faCopy, faArrowDown, faArrowUp, faTrash);

/** The `AddButton` renders a button that represent the `Add` action on a form
 *
 * @param props - The props for the component
 */
export default function AddButton<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any>({
  className,
  onClick,
  disabled,
  registry,
  ...otherProps
}: IconButtonProps<T, S, F>) {
  const { translateString } = registry;
  return (
    <div className='row'>
      <p className={`col-xs-3 col-xs-offset-9 text-right ${className}`}>
        <DaisyUIButton
          {...otherProps}
          iconType='info'
          icon={faPlus as IconDefinition}
          className='btn-add col-xs-12 btn-primary btn-primary-content'
          title={translateString(TranslatableString.AddButton)}
          onClick={onClick}
          disabled={disabled}
          registry={registry}
        />
      </p>
    </div>
  );
}
