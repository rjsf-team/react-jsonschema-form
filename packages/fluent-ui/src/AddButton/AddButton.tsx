import React from "react";

import { AddButtonProps } from "@rjsf/core";

import { IIconProps, CommandBarButton } from "@fluentui/react";

const addIcon: IIconProps = { iconName: "Add" };

const AddButton = (props: AddButtonProps) => (
  <CommandBarButton
    style={{ height: "32px" }}
    iconProps={addIcon}
    text="Add item"
    className={props.className}
    onClick={e => props.onClick(e as any)}
    disabled={props.disabled}
    />

);

export default AddButton;
