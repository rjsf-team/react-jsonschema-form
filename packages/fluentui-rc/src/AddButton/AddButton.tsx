import { FormContextType, IconButtonProps, RJSFSchema, StrictRJSFSchema, TranslatableString } from '@rjsf/utils';
import { Button } from '@fluentui/react-components';
import { AddRegular } from '@fluentui/react-icons';

export default function AddButton<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any>({
  uiSchema,
  registry,
  ...props
}: IconButtonProps<T, S, F>) {
  const { translateString } = registry;
  return <Button title={translateString(TranslatableString.AddItemButton)} {...props} icon={<AddRegular />} />;
}
