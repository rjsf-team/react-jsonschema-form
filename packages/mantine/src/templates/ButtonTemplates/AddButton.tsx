import { FormContextType, IconButtonProps, RJSFSchema, StrictRJSFSchema, TranslatableString } from '@rjsf/utils';
import IconButton from './IconButton';
import { Plus } from '../icons';

export default function AddButton<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any>(
  props: IconButtonProps<T, S, F>
) {
  const {
    registry: { translateString },
  } = props;
  return (
    <IconButton title={translateString(TranslatableString.CopyButton)} variant='subtle' {...props} icon={<Plus />} />
  );
}
