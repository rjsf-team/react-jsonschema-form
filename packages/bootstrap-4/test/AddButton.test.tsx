import React from "react";
import AddButton from "../src/AddButton";
import renderer from "react-test-renderer";

describe("AddButton", () => {
  test("simple", () => {
    const tree = renderer
      .create(
        <AddButton
          className="someClass"
          onClick={() => void 0}
          disabled={false}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
