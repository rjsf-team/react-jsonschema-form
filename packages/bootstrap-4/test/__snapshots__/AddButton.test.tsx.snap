// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddButton simple 1`] = `
<button
  className="ml-1 btn btn-primary"
  color="primary"
  disabled={false}
  onClick={[Function]}
  style={
    Object {
      "width": "100%",
    }
  }
  type="button"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 16 16"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M8 3.5a.5.5 0 01.5.5v4a.5.5 0 01-.5.5H4a.5.5 0 010-1h3.5V4a.5.5 0 01.5-.5z"
      fillRule="evenodd"
    />
    <path
      clipRule="evenodd"
      d="M7.5 8a.5.5 0 01.5-.5h4a.5.5 0 010 1H8.5V12a.5.5 0 01-1 0V8z"
      fillRule="evenodd"
    />
  </svg>
</button>
`;
